src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -220,10 +223,15 @@ private TensorFlowTransform(IHostEnvironment env, byte[] modelBytes, string[] in
                 var tfInput = new TFOutput(Graph[Inputs[i]]);
                 TFInputTypes[i] = tfInput.OutputType;
                 TFInputShapes[i] = Graph.GetTensorShape(tfInput);
-                var newShape = new long[TFInputShapes[i].NumDimensions];
-                for (int j = 0; j < TFInputShapes[i].NumDimensions; j++)
-                    newShape[j] = TFInputShapes[i][j] == -1 ? BatchSize : TFInputShapes[i][j];
-                TFInputShapes[i] = new TFShape(newShape);
+                if (TFInputShapes[i].NumDimensions != -1)
```

Comment:
>if (TFInputShapes[i].NumDimensions != -1) [](start = 16, length = 41)

am curious - why did we need this check ? 

Did some of the pre-trained models have TFInputShapes[i].NumDimensions == -1  (that we were not handling before)

Reply:
If the shape is completely unknown, then its NumDimensions property is -1.

---
In reply to: [216460739](https://github.com/dotnet/machinelearning/pull/857#discussion_r216460739) [](ancestors = 216460739)

Reply:
makes sense. in another comment i was asking if TF has this documented somewhere -- or we are using this as a heuristic based on models we have played with so far ?

---
In reply to: [216475056](https://github.com/dotnet/machinelearning/pull/857#discussion_r216475056) [](ancestors = 216475056,216460739)

Reply:
I haven't seen it documented, I just saw it by debugging different models.

---
In reply to: [216476239](https://github.com/dotnet/machinelearning/pull/857#discussion_r216476239) [](ancestors = 216476239,216475056,216460739)